lib/render/vp2RenderDelegate/proxyRenderDelegate.cpp

```diff
@@ -335,23 +335,25 @@ void ProxyRenderDelegate::_UpdateSceneDelegate()
         _sceneDelegate->SetTime(timeCode);
     }
 
-    {
+    const MMatrix inclusiveMatrix = _proxyDagPath.inclusiveMatrix();
+    const GfMatrix4d transform(inclusiveMatrix.matrix);
+    constexpr double tolerance = 1e-9;
+    if (!GfIsClose(transform, _sceneDelegate->GetRootTransform(), tolerance)) {
         MProfilingScope profilingScope(HdVP2RenderDelegate::sProfilerCategory,
             MProfiler::kColorC_L1, "SetRootTransform");
-
-        // Rprim sync will be triggered only if root transform is changed.
-        const MMatrix inclusiveMatrix = _proxyDagPath.inclusiveMatrix();
-        const GfMatrix4d rootTransform(inclusiveMatrix.matrix);
-        _sceneDelegate->SetRootTransform(rootTransform);
+        _sceneDelegate->SetRootTransform(transform);
     }
 
-    {
+    const bool isVisible = _proxyDagPath.isVisible();
+    if (isVisible != _sceneDelegate->GetRootVisibility()) {
         MProfilingScope profilingScope(HdVP2RenderDelegate::sProfilerCategory,
             MProfiler::kColorC_L1, "SetRootVisibility");
-
-        // Rprim sync will be triggered only if root visibility is changed.
-        const bool isVisible = _proxyDagPath.isVisible();
         _sceneDelegate->SetRootVisibility(isVisible);
+
+        // Trigger selection update when a hidden proxy shape gets shown.
+        if (isVisible) {
```

Comment:
Trigger the selection update when an invisible proxy gets shown. To make this change, I also make the explicit check to see whether there is visibility change (which is checked inside SetRootVisibility() as well).

To make code consistent, I also refactored root transform part, a side benefit of this is the matrix comparison is done by approximation (GfIsClose) rather than equality. This is a TODO item inside SetRootTransform().

FWIW, we shouldn't do similar code change to skip SetTime() call when the time code is equivalent. SetTime() will apply pending edits before its internal time equality check.